src/Avalonia.Controls/Presenters/ItemVirtualizerSimple.cs

```diff
@@ -76,9 +76,23 @@ public override double OffsetValue
                         var firstIndex = ItemCount - panel.Children.Count;
                         RecycleContainersForMove(firstIndex - FirstIndex);
 
-                        panel.PixelOffset = VirtualizingPanel.ScrollDirection == Orientation.Vertical ?
-                            panel.Children[0].Bounds.Height :
-                            panel.Children[0].Bounds.Width;
+                        double pixelOffset;
```

Comment:
@grokys  It might be different it might be the same i'm not sure tough. As i'm unable to debug while this bug happening as it's so rare. Only 2 time i was able to debug the bug in control catalog :(. 
One of the things i've noticed (beside index overflow) is that `child.Bounds` are not guaranteed to be valid (while debugging noticed bounds are 0 once). Probably in measure phase before first arrange. And possibly this can be the initial phase of rubbishing the internal state of the virtualizer. I'm also not sure `DesiredSize` is the proper approach,  but but i think it's better that bounds 0. I can try unit test for this case (no guarantees i'll succeed).

Possible options here can be:

- use Bounds even they are 0

- use DesiredSize if Bounds are 0 (my approach)

- use first child Bounds with valid arrange

- use the average item size which is calculated in the panel somewhere

Let me know what you think is best, but for me `DesiredSize` is the correct approach as it's guaranteed from VirtualizingStackPanel that child has been measured.
https://github.com/AvaloniaUI/Avalonia/blob/8b3c0803c7b6181d80fe8ddb8c08bdfe222d350a/src/Avalonia.Controls/VirtualizingStackPanel.cs#L200-L220

Reply:
Ok, let's go with this solution then!